lbz/response.py

```diff
@@ -17,21 +21,37 @@ def __init__(
         base64_encoded: bool = False,
     ):
         self.body = body
-        self.is_json = isinstance(body, dict)
-        self.headers = headers if headers is not None else self.get_content_header()
+        self.headers = headers if headers is not None else self._get_content_header()
         self.status_code = status_code
+        # TODO: handle bae64 encoded responses appropriately
         self.is_base64 = base64_encoded
+        self._json: dict | None = None
+        if isinstance(body, dict):
+            self._json = body
 
     def __repr__(self) -> str:
         return f"<Response(status_code={self.status_code})>"
 
-    def get_content_header(self) -> dict:
-        """Adds necessary headers based on content type"""
-        if self.is_json:
-            return {"Content-Type": "application/json"}
-        if isinstance(self.body, str):
-            return {"Content-Type": "text/plain"}
-        raise RuntimeError("Response body type not supported yet.")
+    @classmethod
+    def from_exception(cls, error: LambdaFWException, request_id: str) -> Response:
+        """Creates a proper standardised Response for Errors."""
+        resp_data = {"message": error.message, "request_id": request_id}
+        if error.error_code:
+            resp_data["error_code"] = error.error_code
+
+        return cls(resp_data, status_code=error.status_code)
+
+    @property
+    def is_json(self) -> bool:
+        if isinstance(self.body, dict):
+            return True
+        if self.headers and self.headers.get("Content-Type") == "application/json":
```

Comment:
Headers should always be a dictionary, at least initialization does not give any other option:
```suggestion
        if self.headers.get("Content-Type") == "application/json":
```